src/Math-Matrix/PMTensor.class.st

```diff
@@ -0,0 +1,152 @@
+Class {
+	#name : #PMTensor,
+	#superclass : #Object,
+	#instVars : [
+		'array',
+		'shape',
+		'first',
+		'strides'
+	],
+	#category : #'Math-Matrix'
+}
+
+{ #category : #'instance creation' }
+PMTensor class >> fromNestedArray: anArray [
+
+	^ self new array: anArray flattened withShape: (self shape:anArray)
+]
+
+{ #category : #'instance creation' }
+PMTensor class >> newWith: anInteger [
+
+	^ self new array: {anInteger} withShape: #(  )
+]
+
+{ #category : #accessing }
+PMTensor class >> shape: anArray [
+
+anArray isArray ifFalse:[^#()]
+ifTrue:[
+	^ {anArray size}, (self shape: anArray first)
+	]
+
+]
+
+{ #category : #comparing }
+PMTensor >> = aTensor [
+
+	^ array = aTensor array & (first = aTensor first)
+	  & (strides = aTensor strides) & (shape = aTensor shape)
+]
+
+{ #category : #accessing }
+PMTensor >> array [ 
+^array
+]
+
+{ #category : #'as yet unclassified' }
+PMTensor >> array: aFlatArray withShape: aShape [
+
+	array := aFlatArray.
+	shape := aShape copy.
+	self updateFirst.
+	shape ifNotEmpty: [ self updateStrides]
+]
+
+{ #category : #accessing }
+PMTensor >> first [ 
+^first
+]
+
+{ #category : #'primitives - file' }
+PMTensor >> flattenedIndexOf: coords [
+
+	| position |
+	position := 1.
+	coords withIndexDo: [ :elt :i | 
+		position := elt - 1 * (strides at: i) + position ].
+	^ position
+]
+
+{ #category : #'as yet unclassified' }
+PMTensor >> fromNestedArray: aFlatArray withShape: aShape [
+
+	array := aFlatArray.
+	shape := aShape copy.
+	self updateFirst.
+	shape ifNotEmpty: [ self updateStrides]
+]
+
+{ #category : #public }
+PMTensor >> get: coords [
```

Comment:
Is there a good reason for not calling this `at:`?

Reply:
I agree